### PR TITLE
Emulate multi node in shuffle.py

### PR DESCRIPTION
**Pull request by rkooo567**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

**Comment by rkooo567** [comment]
@ericl I added non streaming shuffle here too. Please take a look.

File: python/ray/experimental/shuffle.py [hunk 1]
```diff
@@ -240,19 +256,38 @@ def input_reader(i: PartitionID) -> Iterable[InType]:
...
-        # TODO(ekl) using ray.wait can be more efficient for pipelining.
-        for obj_ref in shuffle_inputs:
-            arr = ray.get(obj_ref)
+        ready, unready = ray.wait(shuffle_inputs)
+        while unready:
+            for obj_ref in ready:
+                arr = ray.get(obj_ref)
+                total += arr.size * arr.itemsize
+            ready, unready = ray.wait(unready)
+            time.sleep(0.1)
```
Comment (ericl) [inline python/ray/experimental/shuffle.py:265]: Can you revert this for now? The behavior could be pretty different. We could also make it a flag.

Comment (ericl) [reply python/ray/experimental/shuffle.py:265]: Also, why the sleep?

File: python/ray/experimental/shuffle.py [hunk 2]
```diff
@@ -64,6 +64,17 @@ def finish(self) -> List[ObjectRef]:
         return self.results
 
 
+class ObjectStoreWriterNonStreaming:
```
Comment (ericl) [inline python/ray/experimental/shuffle.py:67]:
```suggestion
class ObjectStoreWriterNonStreaming(OutputWriter):
```

**Review by ericl — CHANGES_REQUESTED** [review]
Can you revert the wait code?

File: python/ray/experimental/shuffle.py (hunk 1 above)
Comment (ericl) [reply python/ray/experimental/shuffle.py:265]: You can do this instead:
```
while shuffle_inputs:
    [ready], shuffle_inputs = ray.wait(shuffle_inputs, num_returns=1)
    ray.get(ready)
```

Comment (rkooo567) [reply python/ray/experimental/shuffle.py:265]: oh no reason for sleep. I added for debugging and forgot to remove after I fixed a bug.

Comment (rkooo567) [reply python/ray/experimental/shuffle.py]: Added a flag.

**Review by ericl — APPROVED** [review]

**Pull request merged**